File: web/src/dataAttribution.ts
/**
 * dataAttribution.ts
 *
 * Reads bike-network.manifest.json so the credits footer names every agency
 * whose data is actually loaded, with the date it was received.
 *
 * WHY THE DATE MATTERS: Metro's RLIS Limited Use License requires derivative
 * products to display the attribution string AND the date the data was received
 * from Metro. Hardcoding that date in JSX guarantees it goes stale on the next
 * export, so it is read from the manifest the exporter writes (`fetchedAt`).
 *
 * Failure here must never break the app — a missing manifest degrades to the
 * static OSM/PBOT credit that shipped before the metro expansion.
 */

export interface AttributionEntry {
  id: string;
  attribution: string;
  attributionUrl: string;
  fetchedAt: string;
}

interface ManifestEntry extends AttributionEntry {
  file: string;
  features: number;
}

let cached: AttributionEntry[] | null = null;

export async function loadAttributions(): Promise<AttributionEntry[]> {
  if (cached) return cached;
  try {
    const res = await fetch("/bike-network.manifest.json");
    if (!res.ok) return [];
    const manifest = (await res.json()) as ManifestEntry[];
    cached = manifest
      .filter((m) => m.attribution)
      .map(({ id, attribution, attributionUrl, fetchedAt }) => ({
        id,
        attribution,
        attributionUrl,
        fetchedAt,
      }));
    return cached;
  } catch {
    return [];
  }
}

File: ios/BikeRouteNicePDX/Services/DataAttribution.swift
import Foundation

/// Credits for the bundled bike-network data, read from bike-network.manifest.json.
///
/// The app previously credited nothing at all — not OSM, not PBOT — which OSM's
/// ODbL already required and which App Store review flags. Metro's RLIS license
/// goes further: it requires its attribution string verbatim AND the date the
/// data was received, so that date is read from the manifest the exporter writes
/// (`fetchedAt`) rather than hardcoded here, where it would go stale on the next
/// export.
///
/// Mirrors web/src/dataAttribution.ts.
struct DataAttribution: Decodable, Identifiable {
    let id: String
    let attribution: String
    let attributionUrl: String
    let fetchedAt: String

    /// "© Oregon Metro (2026-08-06)"
    var display: String {
        fetchedAt.isEmpty ? attribution : "\(attribution) (\(fetchedAt))"
    }

    var url: URL? { URL(string: attributionUrl) }
}

enum DataAttributionLoader {
    /// Bundled manifest entries, or an empty list if the manifest is missing or
    /// malformed — credits must never be the thing that crashes the app.
    static func load() -> [DataAttribution] {
        guard
            let url = Bundle.main.url(forResource: "bike-network.manifest", withExtension: "json"),
            let data = try? Data(contentsOf: url),
            let entries = try? JSONDecoder().decode([DataAttribution].self, from: data)
        else {
            return []
        }
        return entries.filter { !$0.attribution.isEmpty }
    }
}
